### DISCUSSION TITLE
Dishwasher device and entity name

### DISCUSSION THREAD

**Discussion by storebatfar** [ekutner/home-connect-hass#421] [Community Support]
Hey guys.
I have just installed my new Siemens dishwasher and connected it to HomeConnect.
It's now showing up in Home Assistant, but the device and entity names are only numbers.
It receives the name '014080389763001121' and the entities are names eg. 'sensor.014080389763001121_active_program'.

My other devices like my washing machine and dryer are named like this: sensor.bosch_waxh2e0lsn_68a40e9322f8_active_program

Is there anything I can do about this?

**Comment by eMeF1** [comment]
@storebatfar I just went through the same process. Here’s what I did:

1. Navigate to **Integration > Home Connect Alt** and open **Devices** (click on the "_n_ devices" link below the Integration entities).
2. Select the desired device.
3. Change the device name:
   - Click the pencil icon in the top right corner.
   - Set the device name to something reasonable (e.g., **Washing Machine**) and click **Update**.
4. Under **Controls**, click on the icon of the first entity.
5. Click on the **Cog Wheel** to edit the entity.
   - Update the name to follow the format **[Device Name] [Entity Description]** (e.g., **Washing Machine Childproof lock**).
     - Ensure the name starts with the Device Name, followed directly by the Entity Description without a dash or other punctuation.
     - Home Assistant will remove the Device Name in the device dashboard and will display only the Entity Description.
   - Optionally, change the Entity ID to **[device_name]_[entity_id]** (e.g., `switch.washing_machine_childlock`).
   - Click **Update** to apply changes.
6. Repeat this process for all entity Controls and Sensors.

Somewhat cumbersome, but the result is quite polished.


<img width="1477" alt="Screenshot 2024-12-19 at 7 33 47 PM" src="https://github.com/user-attachments/assets/79090b5f-927a-4787-908b-e59a47bc745e" />
